File: core/src/main/kotlin/com/rarible/blockchain/scanner/EnableBlockchainScanner.java
package com.rarible.blockchain.scanner;

import com.rarible.blockchain.scanner.configuration.BlockchainScannerConfiguration;
import org.springframework.context.annotation.Import;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Import(BlockchainScannerConfiguration.class)
public @interface EnableBlockchainScanner {

}

File: core/src/main/kotlin/com/rarible/blockchain/scanner/EnableBlockchainScanner.kt
package com.rarible.blockchain.scanner

import com.rarible.blockchain.scanner.configuration.BlockchainScannerConfiguration
import org.springframework.context.annotation.Import

@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Import(BlockchainScannerConfiguration::class)
annotation class EnableBlockchainScanner
